prow/pod-utils/decorate/podspec.go

```diff
@@ -0,0 +1,311 @@
+			},
+		}
+
+		toolsMount := kube.VolumeMount{
+			Name:      ToolsMountName,
+			MountPath: ToolsMountPath,
+		}
+		toolsVolume := kube.Volume{
+			Name: ToolsMountName,
+			VolumeSource: kube.VolumeSource{
+				EmptyDir: &kube.EmptyDirVolumeSource{},
+			},
+		}
+
+		gcsCredentialsMount := kube.VolumeMount{
+			Name:      GcsCredentialsMountName,
+			MountPath: GcsCredentialsMountPath,
+		}
+		gcsCredentialsVolume := kube.Volume{
+			Name: GcsCredentialsMountName,
+			VolumeSource: kube.VolumeSource{
+				Secret: &kube.SecretSource{
+					SecretName: pj.Spec.GCSConfiguration.CredentialsSecret,
+				},
+			},
+		}
+
+		cloneLogFlag := fmt.Sprintf("--log=%s/clone.json", LogMountPath)
+		gcsFlags := []string{
+			"--dry-run=false",
+			fmt.Sprintf("--gcs-bucket=%s", pj.Spec.GCSConfiguration.Bucket),
+			fmt.Sprintf("--gcs-credentials-file=%s/service-account.json", GcsCredentialsMountPath),
+			fmt.Sprintf("--path-strategy=%s", pj.Spec.GCSConfiguration.PathStrategy),
+			fmt.Sprintf("--default-org=%s", pj.Spec.GCSConfiguration.DefaultOrg),
+			fmt.Sprintf("--default-repo=%s", pj.Spec.GCSConfiguration.DefaultRepo),
+		}
+
+		spec.InitContainers = []kube.Container{
+			{
+				Name:    "clonerefs",
+				Image:   pj.Spec.UtilityImages.CloneRefs,
+				Command: []string{"/clonerefs"},
+				Args: []string{
+					fmt.Sprintf("--src-root=%s", CodeMountPath),
+					cloneLogFlag,
+				},
+				Env:          env,
+				VolumeMounts: []kube.VolumeMount{logMount, codeMount},
+			},
+			{
+				Name:         "initupload",
+				Image:        pj.Spec.UtilityImages.InitUpload,
+				Command:      []string{"/initupload"},
+				Args:         append(gcsFlags, cloneLogFlag),
+				Env:          env,
+				VolumeMounts: []kube.VolumeMount{logMount, gcsCredentialsMount},
+			},
+			{
+				Name:    "place-tools",
+				Image:   pj.Spec.UtilityImages.Entrypoint,
+				Command: []string{"/bin/cp"},
+				Args: []string{
+					"/entrypoint",
+					fmt.Sprintf("%s/entrypoint", ToolsMountPath),
+				},
+				VolumeMounts: []kube.VolumeMount{toolsMount},
+			},
+		}
+
+		processFlags := []string{
+			fmt.Sprintf("--process-log=%s/process-log.txt", LogMountPath),
+			fmt.Sprintf("--marker-file=%s/marker-file.txt", LogMountPath),
+		}
+
+		testArgs := append(spec.Containers[0].Command, spec.Containers[0].Args...)
+		spec.Containers[0].Command = []string{"/tools/entrypoint"}
+		spec.Containers[0].Args = append(
+			processFlags,
+			append([]string{
+				fmt.Sprintf("--timeout=%s", (time.Duration(pj.Spec.TimeoutMinutes) * time.Minute).String()),
+				"--",
+			}, testArgs...)...)
+		if pj.Spec.Type != kube.PeriodicJob {
+			clonePath := fmt.Sprintf("github.com/%s/%s", pj.Spec.Refs.Org, pj.Spec.Refs.Repo)
+			if pj.Spec.Refs.PathAlias != "" {
+				clonePath = pj.Spec.Refs.PathAlias
+			}
+			spec.Containers[0].WorkingDir = fmt.Sprintf("%s/src/%s", CodeMountPath, clonePath)
+		}
+		spec.Containers[0].VolumeMounts = append(spec.Containers[0].VolumeMounts, logMount, codeMount, toolsMount)
```

Comment:
We need to determine how we are going to inform the test process of the `/logs` mount where it should dump artifacts -- I think right now `$WORKSPACE/_artifacts` is used?

Reply:
I think we should have an env specifically for this like `$ARTIFACTS_DIR` so we can refactor this at any time and simplify the code consuming this